shotover-proxy/tests/cassandra_int_tests/mod.rs

```diff
@@ -662,6 +663,70 @@ async fn request_throttling(#[case] driver: CassandraDriver) {
     shotover.shutdown_and_then_consume_events(&[]).await;
 }
 
+#[rstest]
+#[case::cdrs(CdrsTokio)]
+#[case::scylla(Scylla)]
+#[tokio::test(flavor = "multi_thread")]
+#[serial]
+async fn compression_single(#[case] driver: CassandraDriver) {
+    async fn test(driver: CassandraDriver, topology_path: &str, compression: Compression) {
+        let _compose =
+            DockerCompose::new("example-configs/cassandra-passthrough/docker-compose.yaml");
+        let shotover = shotover_from_topology_file(topology_path).await;
+        let connection =
+            || CassandraConnection::new_with_compression("127.0.0.1", 9042, driver, compression);
+
+        standard_test_suite(connection, driver).await;
+
+        shotover.shutdown_and_then_consume_events(&[]).await;
+    }
+
+    // passthrough
+    for topology in [
+        "example-configs/cassandra-passthrough/topology.yaml",
+        "example-configs/cassandra-passthrough/topology-encode.yaml",
+    ] {
+        for compression in [Compression::Lz4, Compression::Snappy] {
+            test(driver, topology, compression).await;
+        }
+    }
+}
+
```

Comment:
> Not convinced covering each compression type with passthrough/encoding on each driver is necessary, if reviewers agree I can make the tests smaller.

On my machine it only takes 3mins total to run these new tests.
But it seems to take 20mins on CI which is pretty bad!
So I agree, lets trim it down a bit, I would probably get rid of cdrs on both tests (I dont think testing cdrs is critical) and get rid of LZ4 on compression_cluster (compression_cluster is slower, so theres a bigger win to remove from there)